docs/program/extend/modular-resources/_index.md

```diff
@@ -1,150 +1,528 @@
+- Create new models of built-in component or service types
 - Create brand new resource types
 
-Modular resources that are configured and composed into a Viam robot behave identically to built-in resources.
-This means that functionality that the Viam [RDK](/internals/rdk/) provides for built-in resources is also automatically provided for user-created resources.
+`viam-server` [manages](#modular-resource-management) modular resources configured on your robot like resources that are already built-in to the [Robot Development Kit (RDK)](/internals/rdk).
+
+Two key concepts exist across all Viam resources (both built-in and modular) to facilitate this: [*APIs*](#apis) and [*models*](#models).
 
-Two key concepts exist across all Viam resources (built-in and modular) to facilitate this: *APIs* and *models*
+## Key concepts
 
 ### APIs
 
-Every Viam component and service type exposes an *API*.
-This API describes the interface for the particular component or service type.
-For example, the API of built-in component type [camera](/components/camera) exposes methods such as *GetImage*, and the API of built-in service type [vision](/services/vision) exposes methods such as *GetDetectionsFromCamera*.
+Every Viam [resource](/appendix/glossary/#term-resource) exposes an [Application Programming Interface (API)](https://en.wikipedia.org/wiki/API).
+This can also be understood as the protocol that the resource "speaks".
+Each API is described through <a href="https://developers.google.com/protocol-buffers" target="_blank">protocol buffers</a>.
+Viam SDKs [expose these APIs](/internals/robot-to-robot-comms/).
 
-Each API corresponds with a Viam resource and is described through <a href="https://developers.google.com/protocol-buffers" target="_blank">protocol buffers</a>.
-You can see built-in Viam resource APIs in the <a href="https://github.com/viamrobotics/api" target="_blank">Viam API GitHub repository</a>.
+Each Viam resource's API is uniquely namespaced as a colon-delimited-triplet in the form of `namespace:type:subtype`.
 
-[Viam SDKs](/program/sdk-as-client/) that expose these APIs are available in various programming languages and allow you to control your robots [securely from anywhere](/internals/robot-to-robot-comms/).
+For example:
 
-Viam APIs are uniquely namespaced, with each resource type represented as a *colon-delimited-triplet*.
-For example, the built-in component type *camera*'s API is __rdk:component:camera__.
-The built-in service type *vision*'s API is __rdk:service:vision__
+- The API of built-in component [camera](/components/camera) is `rdk:component:camera`, which exposes methods such as `GetImage()`.
+- The API of built-in service [vision](/services/vision) is `rdk:service:vision`, which exposes methods such as `GetDetectionsFromCamera()`.
+
+{{% alert title="Note" color="note" %}}
+You can see built-in Viam resource APIs in the <a href="https://github.com/viamrobotics/api" target="_blank">Viam GitHub</a>.
```

Comment:
why is this not using mardown-style links?

Reply:
Just left over from og, will fix